Microsoft Azure Active Directory SSO Instructions

stevenGravy opened the pull request:
How to documentation on creating a SAML connector to Micorsoft Azure Active Directory (AD) saml connector.

stevenGravy committed:
Create ssh_azuread.md

stevenGravy committed:
Added Azure Active Directory link

stevenGravy committed:
update graphics

stevenGravy committed:
AzureAD images

stevenGravy committed:
Update ssh_azuread.md

stevenGravy committed:
Add files via upload

stevenGravy committed:
Update ssh_azuread.md

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+based access control (RBAC) it allows SSH administrators to define policies
+like:
+
+* Only members of "DBA" Google group can SSH into machines running PostgreSQL.
```
Google group? This is AD

stevenGravy replied:
Changed to Msft

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+
+!!! warning "Version Warning"
+
+    This guide requires an enterprise version of Teleport. The open source
```
enterprise -> Enterprise

stevenGravy replied:
changed

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+
+- An Enterprise version of Teleport v4.2 or greater, downloaded from [https://dashboard.gravitational.com/](https://dashboard.gravitational.com/web/). 
+- Azure AD admin account with access to creating non-gallery applications (P2 License)
+- Register one or more users in the directory
```
Register -> To register (because of the "you'll need" header)

stevenGravy replied:
changed

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+- An Enterprise version of Teleport v4.2 or greater, downloaded from [https://dashboard.gravitational.com/](https://dashboard.gravitational.com/web/). 
+- Azure AD admin account with access to creating non-gallery applications (P2 License)
+- Register one or more users in the directory
+- Create at least two security groups in AzureAD and assign one or more users to each group
```
Create -> To create (because of the header)

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+3. Select a Non-gallery application
+   ![Select Non-gallery application](img/azuread/azuread-3-selectnongalleryapp.png)
+   
+4. Enter the display name (Ex: teleport)
```
teleport -> Teleport

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+6. Select Single Sign-on under Manage and choose SAML
+   ![Select SAML](img/azuread/azuread-6-selectsaml.png)
+   
+7. Select to edit  Basic SAML Configuration
```
double space between edit and Basic

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+7. Select to edit  Basic SAML Configuration
+   ![Edit Basic SAML Configuration](img/azuread/azuread-7-editbasicsaml.png)
+   
+8. Put in the Entity ID and Reply URL the same proxy url https://proxy1:3080/v1/webapi/saml/acs
```
@benarent Do we have a default URL we use for replacements here? like https://<proxy-address> or whatever

stevenGravy replied:
using the same example teleport.example.com

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+   
+9. Edit User Attributes & Claims
+
+a. Edit the Claim Name Change the name identifier format to Default Make sure the source attribute is user.userprincipalname
```
This sentence needs some punctuation

stevenGravy replied:
fixed

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+a. Edit the Claim Name Change the name identifier format to Default Make sure the source attribute is user.userprincipalname
+   ![Confirm Name Identifier](img/azuread/azuread-9a-nameidentifier.png)
+   
+b. Add a group Claim to have user security groups groups
```
groups groups?

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+b. Add a group Claim to have user security groups groups
+   ![Put in Security group claim](img/azuread/azuread-9b-groupclaim.png)
+   
+10. On the SAM Signing Certificate selec to SAML Download the Federation Metadata XML.  
```
SAM -> SAML
selec -> select

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+10. On the SAM Signing Certificate selec to SAML Download the Federation Metadata XML.  
+   ![Download Federation Metadata XML](img/azuread/azuread-10-fedmeatadataxml.png)
+
+!!! warning "Important"  This a imporant document.  Treat this file as you would a password.
```
imporant -> important

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+
+## Create a SAML Connector
+
+Now, create a SAML connector [resource](admin-guide.md#resources).  Replace the <proxy address> with your teleport address and the group ids with the AzureAD group values.
```
teleport -> Teleport
ids -> IDs

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+  display: "Microsoft"
+  # acs is the Assertion Consumer Service URL. This should be the address of
+  # the Teleport proxy that your identity provider will communicate with.
+  acs: https://<proxy address>:3080/v1/webapi/saml/acs
```
Again not sure if we have a default string to replace here for proxy address, if we don't we should use the same string throughout the whole doc

stevenGravy replied:
using the teleport.example.com

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+
+We are going to create 2 roles:
+
+-  Privileged role admin who is able to login as root and is capable of administrating
```
admin -> `admin` maybe to make it clearer that it's the role name

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+
+-  Privileged role admin who is able to login as root and is capable of administrating
+the cluster
+- Non-privileged dev
```
dev -> role `dev`

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+  options:
+    max_session_ttl: 24h
+  allow:
+    logins: [ "{% raw %}{{external.username}}{% endraw %}", ubuntu ]
```
I have feeling that this is a longer string when coming from AD? See https://gravitational.com/teleport/docs/ssh_adfs/#create-teleport-roles

Something like `logins: [ '{{external."http://schemas.microsoft.com/ws/2008/06/identity/claims/windowsaccountname"}}', ubuntu ]`

stevenGravy replied:
I provided instructions on how to make a shorter username with a claim

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+version: v2
+metadata:
+  # the name of the connector
+  name: saml
```
I would probably give this another name like "azure-saml" or something just in case they have multiple connectors

stevenGravy replied:
changed

webvictim commented on docs/4.2/ssh_azuread.md:
````diff
@@ -0,0 +1,190 @@
+the same as before:
+
+```bsh
+$ tsh --proxy=proxy.example.com login
````
This only works if you change your Teleport config:

```yaml
auth_service:
  authentication:
    type: saml
```

Otherwise you need to use`tsh --proxy=proxy.example.com --auth=azure-saml login`

stevenGravy replied:
provided instructions on changing the default authentication type

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,190 @@
+
+!!! tip "Tip"
+
+    Teleport can use multiple OIDC connectors. In this case a connector name
```
OIDC -> OIDC/SAML (as this is a SAML guide)

stevenGravy replied:
changed to saml

webvictim reviewed (changes requested):
Like this! Have some nits/changes but otherwise great

stevenGravy committed:
Update ssh_azuread.md

stevenGravy committed:
Add files via upload

stevenGravy committed:
Update ssh_azuread.md

stevenGravy committed:
Update ssh_azuread.md

stevenGravy committed:
Update ssh_azuread.md

stevenGravy committed:
Update ssh_azuread.md

stevenGravy committed:
Update ssh_azuread.md

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,204 @@
+
+## Create a SAML Connector
+
+Now, create a SAML connector [resource](admin-guide.md#resources).  Replace the acs element with your Teleport address, update the group IDs with the actual AzureAD group ID values, and insert the downloaded Federation Metadata XML into the .
```
Sentence doesn't finish - "into the"...

stevenGravy replied:
corrected sentence

webvictim commented on docs/4.2/ssh_azuread.md:
````diff
@@ -0,0 +1,204 @@
+```bsh
+$ tctl create azure-connector.yaml
+```
+!!!Note Teleport will automatically transform the contents of the connector when viewed from the web UI.
````
I think you have to format this as `!!! note` for `mkdocs` to pick it up and transform it.

You may also need a line break between it and the code block above.

stevenGravy replied:
modified to a tip with the right format.

webvictim commented on docs/4.2/ssh_azuread.md:
```diff
@@ -0,0 +1,204 @@
+## Testing
+
+
+Update the teleport settings to use the saml settings to make this the default.
```
teleport -> Teleport

saml -> SAML

stevenGravy replied:
corrected

stevenGravy committed:
updates for review

Changed note to tip in proper format, corrected sentence and proper cap for Teleport, SAML.

stevenGravy committed:
Merge branch 'master' into stevenGravy-azuread-docs-1

webvictim commented:
LGTM
